install.js

```diff
@@ -20,105 +20,147 @@
+      console.error(error);
+      process.exit(1);
+    }
+
+    let progressBar = null;
+    let lastDownloadedBytes = 0;
+    function onProgress(downloadedBytes, totalBytes) {
+      if (!progressBar) {
+        const ProgressBar = require('progress');
+        progressBar = new ProgressBar(`Downloading ${supportedProducts[product]} r${revision} - ${toMegabytes(totalBytes)} [:bar] :percent :etas `, {
+          complete: '=',
+          incomplete: ' ',
+          width: 20,
+          total: totalBytes,
+        });
+      }
+      const delta = downloadedBytes - lastDownloadedBytes;
+      lastDownloadedBytes = downloadedBytes;
+      progressBar.tick(delta);
+    }
+  }
 
-/**
- * @param {!Array<string>}
- * @return {!Promise}
- */
-function onSuccess(localRevisions) {
-  logPolitely('Chromium downloaded to ' + revisionInfo.folderPath);
-  localRevisions = localRevisions.filter(revision => revision !== revisionInfo.revision);
-  // Remove previous chromium revisions.
-  const cleanupOldVersions = localRevisions.map(revision => browserFetcher.remove(revision));
-  return Promise.all([...cleanupOldVersions, generateProtocolTypesIfNecessary(true /* updated */)]);
-}
+  function toMegabytes(bytes) {
+    const mb = bytes / 1024 / 1024;
+    return `${Math.round(mb * 10) / 10} Mb`;
+  }
 
-/**
- * @param {!Error} error
- */
-function onError(error) {
-  console.error(`ERROR: Failed to download Chromium r${revision}! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.`);
-  console.error(error);
-  process.exit(1);
-}
+  function generateProtocolTypesIfNecessary(updated, product) {
+    if (product !== 'chrome')
+      return;
+    const fs = require('fs');
+    const path = require('path');
+    if (!fs.existsSync(path.join(__dirname, 'utils', 'protocol-types-generator')))
+      return;
+    if (!updated && fs.existsSync(path.join(__dirname, 'lib', 'protocol.d.ts')))
+      return;
+    return require('./utils/protocol-types-generator');
+  }
 
-let progressBar = null;
-let lastDownloadedBytes = 0;
-function onProgress(downloadedBytes, totalBytes) {
-  if (!progressBar) {
-    const ProgressBar = require('progress');
-    progressBar = new ProgressBar(`Downloading Chromium r${revision} - ${toMegabytes(totalBytes)} [:bar] :percent :etas `, {
-      complete: '=',
-      incomplete: ' ',
-      width: 20,
-      total: totalBytes,
+  function getFirefoxNightlyVersion(host) {
+    const https = require('https');
+    const promise = new Promise((resolve, reject) => {
+      let data = '';
+      logPolitely(`Requesting latest Firefox Nightly version from ${host}`);
+      https.get(host + '/', r => {
+        r.on('data', chunk => {
+          data += chunk;
+        });
+        r.on('end', parseVersion);
+      }).on('error', e => reject(e));
+
+      function parseVersion() {
+        const regex = /firefox\-(?<version>\d\d)\..*/gm;
```

Comment:
Loving the use of named capture groups here!

This will fail on Node.js v8.x.x so we'd want to land #https://github.com/puppeteer/puppeteer/pull/5365 first. @hanselfmu could you please take another look?

Reply:
The PR to drop Node.js v8.x support is labeled as a breaking change, which means landing it will be in Puppeteer v3. I think we have two options now:
1. land both of them in v3;
2. temporarily use Node.js-8-compatible code for now, and when v3 lands, update the legacy code. 
@mjzffr what do you think? Do you have other ideas?

Reply:
IMHO major version bumps are no big deal. We shouldn't hold back on publishing v3.

Reply:
Sounds good. I'll publish v3 now and include this PR.

Reply:
@hanselfmu @mathiasbynens What happened with this? This PR doesn't seem to be available in a release version yet.